src/azul/plugins/metadata/hca/transform.py

```diff
@@ -81,6 +85,123 @@
+        >>> a.to_index(None)
+        '~null'
+
+        >>> a.to_index({'notval': '20', 'notu': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A dictionary with entries for `value` and `unit` is required
+
+        >>> a.to_index({'value': '', 'unit': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not be empty
+
+        >>> a.to_index({'value': '20', 'unit': ''})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not be empty
+
+        >>> a.to_index({'value': None, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Use `null` for entities with no organism age
+
+        >>> a.to_index({'value': '20 ', 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not contain space characters
+
+        >>> a.to_index({'value': 20, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must be of string type
+
+        >>> a.to_index({'value': '20', 'unit': 'years '})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not contain space characters
+
+        >>> a.to_index({'value': '20', 'unit': True})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must be of string type
+
+        """
+        if value is None:
+            return NullableString.null_string
+        else:
+            try:
+                value, unit = value['value'], value['unit']
+            except KeyError:
+                raise RequirementError(
+                    'A dictionary with entries for `value` and `unit` is required'
+                )
+            else:
+                reject(value == '', 'The `value` entry must not be empty')
+                reject(unit == '', 'The `unit` entry must not be empty')
+        if value is None:
+            raise RequirementError(
+                'Use `null` for entities with no organism age'
+            )
+        elif unit is None:
+            return value
+        else:
+            require(type(value) is str, 'The `value` entry must be of string type')
+            require(type(unit) is str, 'The `unit` entry must be of string type')
+            reject(' ' in value, 'The `value` entry must not contain space characters')
+            reject(' ' in unit, 'The `unit` entry must not contain space characters')
+            return f'{value} {unit}'
+
+    def from_index(self, value: str) -> Optional[JSON]:
+        """
+        >>> a = OrganismAge()
+        >>> a.from_index('20 year')
+        {'value': '20', 'unit': 'year'}
+
+        >>> a.from_index('20')
+        {'value': '20', 'unit': None}
+
+        >>> a.from_index('~null')
```

Comment:
```suggestion
        >>> a.from_index('~null') is None
        True
```

is a bit more explicit